File: web_app/getClassNames.js
let classNames = [];

document.getElementById('fileInput').addEventListener('change', function(event) {
    classNames = [];
    const fileList = event.target.files; // List of dataset filenames selected
    
    for (let i = 0; i < fileList.length; i++) {
      const file = fileList[i];
      let fileName = file.name;
      // Format the filename for clarity in the class name
      fileName = fileName.replace('full_numpy_bitmap_', '').replace('.npy', '');
      fileName = fileName.charAt(0).toUpperCase() + fileName.slice(1);
      classNames.push(fileName);
    }

    // Notify that the user has selected their datasets
    document.dispatchEvent(new Event('classNamesUpdated'));
});

File: python_model/getClassNames.py
import os

DATASET_LIST = os.listdir('../QuickDrawDataset')
classNames = []

# Formats each dataset filename to get a list of class names
for item in DATASET_LIST:
    item = item.strip()
    classNames.append(item.replace('full_numpy_bitmap_', '').replace('.npy', '').title())